public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -141,23 +139,29 @@ export class ResultTransformer {
 
   transformInstantMetricData(md: any, options: any) {
     const dps = [];
-    let metricLabel = null;
-    metricLabel = this.createMetricLabel(md.metric, options);
+    const { name, labels } = this.createLabelInfo(md.metric, options);
     dps.push([parseFloat(md.value[1]), md.value[0] * 1000]);
-    return { target: metricLabel, datapoints: dps, tags: md.metric, refId: options.refId };
+    return { target: name, datapoints: dps, tags: labels, refId: options.refId };
   }
 
-  createMetricLabel(labelData: { [key: string]: string }, options: any) {
-    let label = '';
-    if (_.isUndefined(options) || _.isEmpty(options.legendFormat)) {
-      label = this.getOriginalMetricName(labelData);
-    } else {
-      label = this.renderTemplate(this.templateSrv.replace(options.legendFormat), labelData);
+  createLabelInfo(labelData: { [key: string]: string }, options: any): { name: string; labels: Labels } {
+    if (options?.legendFormat) {
+      // Alternativly we could put the results in "title"
+      const name = this.renderTemplate(this.templateSrv.replace(options.legendFormat), labelData);
+      if (name) {
+        return { name, labels: labelData };
+      }
     }
```

Comment:
So one remaining problem now is if you set a legend format, we preserve the labels. So getFieldDisplayText does not know that a legend format has been used and so you end up with legend format result and labels 
<img width="1219" alt="Screen Shot 2020-04-30 at 12 00 34" src="https://user-images.githubusercontent.com/10999/80698874-5d89bc00-8adb-11ea-9222-d96d632898f4.png">